jax/interpreters/xla.py

```diff
@@ -1021,7 +1021,8 @@ def f_with_avals(c, avals, xla_args, params):
     wrapped_fun = lu.wrap_init(fun, params)
     if not multiple_results:
       wrapped_fun = _tuple_output(wrapped_fun)
-    jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(wrapped_fun, avals)
+    with core.extend_axis_env_nd(zip(axis_env.names, axis_env.sizes)):
```

Comment:
Shouldn't we extend the env in `api.xla_computation` instead?

Reply:
Good question. Thanks for checking this.

[We already extend the env while forming the jaxpr](https://github.com/google/jax/blob/fe35680d1844f063d396e3e376c3dad500b922d7/jax/_src/api.py#L683-L686), and there's no need to populate the env [when doing jaxpr-to-hlo lowering](https://github.com/google/jax/blob/fe35680d1844f063d396e3e376c3dad500b922d7/jax/_src/api.py#L699) (since we're no longer tracing / calling bind, so nothing should need the trace-time env...) _unless_ we call into `xla.lower_fun`. For that reason I think this is the right place to extend the axis env.

WDYT?

Reply:
I'll copy some explanatory comments I made in JAXers chat into the PR message above.

Reply:
(I speculatively merged, but if we fiugre out that this is the wrong approach I will revise it!)